### ISSUE TITLE
Reverse escape code with transparency draws foreground without transparency

### ISSUE THREAD

**Issue by noriah** [alacritty/alacritty#6433]
When using the reverse drawing escape code (`ESC[7m`) combined with a transparent background, the foreground is drawn using the terminal background color without transparency.

![image](https://user-images.githubusercontent.com/1609931/197262815-55b1e4d9-c5eb-4d35-bc36-daaa1bf70a3a.png)

I am using catnip which uses the subdivided box characters (U+2581 - U+258F) to draw the bar caps. The bottom caps are drawn by using the reverse drawing escape code and the default color for foreground and background.

The current behavior makes some sense. Transparency is applied to the background, so foreground would not be affected. But since the point of the reverse code is to swap the colors of foreground and background, it also makes sense for the foreground to inherit the transparency.

### System

OS: macOS
Version: 0.11.0 (8dbaa0b)

### Config

```yml
window:
  dynamic_padding: true
  decorations: buttonless
  opacity: 0.0

scrolling:
  history: 0

font:
  normal:
    family: FiraCode Nerd Font

  size: 6.5
  builtin_box_drawing: false

colors:
  primary:
    background: '#272936'
    foreground: '#f7f7f7'

  normal:
    white:   '#f7f7f7'

  transparent_background_colors: false
```

**Comment by kchibisov** [comment]
`transparent_background_colors` set to `true` is what you want?

Though it might be desired to also make what you say as transparent without the option.

**Comment by chrisduerr** [comment]
Working as expected.

**Comment by noriah** [comment]
@kchibisov From my testing, that parameter does not change the behavior I am describing. The foreground seems to get past it.

> Though it might be desired to also make what you say as transparent without the option.

Can you rephrase that? I don't understand what you mean.
